src/connection_string.ts

```diff
@@ -1097,15 +1096,29 @@ export const OPTIONS = {
     }
   },
   tlsCAFile: {
-    target: 'ca',
-    transform({ values: [value] }) {
-      return fs.readFileSync(String(value), { encoding: 'ascii' });
+    transform({ name, values: [value] }) {
+      if (typeof value !== 'string') {
+        throw new MongoParseError(`${name} must be of type string`);
+      }
+
+      if (value.length === 0) {
+        throw new MongoParseError(`${name} must be have non-zero length`);
+      }
```

Comment:
This changes the validation a bit more than I would expect from what we had before. I do think we should remove the string coercion that was present here before, it unnecessarily limited the inputs that `fs.readFile` accepts. Can we lean on Node.js errors for both the case of incorrect type and incorrect length? The other indication that we're overdoing it here is that even Node.js does not validate 0 length strings, it is the `open` system call returning `ENOENT` (`errno == -2`) that indicates an issue.

Reply:
So as far as moving forward here, it seems that we only want to pass through strings to the `fs.readFile` call, but I'm not clear on whether or not we want to throw on options parsing or at connect time and whether or not we want to check in our code whether or not the file path is empty.

Reply:
I think the known breakage here was that the error was moving to connect time, but then we're introducing new validation to construct time (currently). I think we should instead pass along whatever input we're given to the FS API.